library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -517,6 +552,18 @@ public void teardown()
   }
 
   @Override
+  public void activate(OperatorContext context)
+  {
+    scanDirService.scheduleAtFixedRate(new DirectoryScannerThread(), 0, scanIntervalMillis, TimeUnit.MILLISECONDS);
```

Comment:
Consider the following scenario: Each DirectoryScannerThread takes 10s and scanIntervalMillis has the default value of 5000. The queue of tasks will continue to grow in an unbounded way. Can we check if the queue is empty before enqueueing a new task ?